### ISSUE TITLE
dhcpcd broken

### ISSUE THREAD

**Issue by dotlambda** [NixOS/nixpkgs#34740]
## Issue description
dhcpcd is broken on master

See also https://hydra.nixos.org/build/68820765

### Steps to reproduce
```
$ nix-build -A dhcpcd
these derivations will be built:
  /nix/store/kkg2xrly7hlfccm539q7zypnpg9qmjx9-dhcpcd-6.11.5.drv
building '/nix/store/kkg2xrly7hlfccm539q7zypnpg9qmjx9-dhcpcd-6.11.5.drv'...
unpacking sources
unpacking source archive /nix/store/2f173jk2hwsbl8mhc5xsf1i3rciy01hx-dhcpcd-6.11.5.tar.xz
source root is dhcpcd-6.11.5
setting SOURCE_DATE_EPOCH to timestamp 1476000437 of file dhcpcd-6.11.5/test/test_hmac_md5.c
patching sources
configuring
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/cz3mn469fw248idhf0wlk7vq9z1xgd54-dhcpcd-6.11.5 --sysconfdir=/etc --localstatedir=/var
configure args: --disable-static --disable-dependency-tracking --prefix=/nix/store/cz3mn469fw248idhf0wlk7vq9z1xgd54-dhcpcd-6.11.5 --sysconfdir=/etc --localstatedir=/var
Deriving operating system from ... x86_64-unknown-linux
Configuring dhcpcd for ... linux
./configure: line 259: syntax error: support for $((arith)) is disabled
builder for '/nix/store/kkg2xrly7hlfccm539q7zypnpg9qmjx9-dhcpcd-6.11.5.drv' failed with exit code 2
error: build of '/nix/store/kkg2xrly7hlfccm539q7zypnpg9qmjx9-dhcpcd-6.11.5.drv' failed
```

## Technical details
 - system: `"x86_64-linux"`
 - host os: `Linux 4.15.1, NixOS, 18.03.git.355de06 (Impala)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.0pre5889_c287d731`
 - channels(root): `"nixos-18.03pre126729.2e4aded3669"`
 - channels(rschuetz): `""`
 - nixpkgs: `/nix/var/nix/profiles/per-user/root/channels/nixos/nixpkgs

/cc @edolstra @fpletz

**Comment by edolstra** [comment]
This was a sandbox issue. Fixed.

**Comment by dotlambda** [comment]
I don't think this is fixed. I get the issue on both, master and staging.

**Comment by dotlambda** [comment]
Okey, now it seems to be building. Don't know what went wrong there. Thank you!

**Comment by gebner** [comment]
This issue seems to be back and breaks the unstable channels:
 - https://hydra.nixos.org/build/71760399/nixlog/1
 - https://hydra.nixos.org/build/71644950/nixlog/1

**Comment by vcunat** [comment]
Should be fixed since cb196f490eb.  At least on my machine it got fixed by that commit.
